lib/core/viewer/components/img-viewer.component.ts

```diff
@@ -55,128 +56,96 @@ export class ImgViewerComponent implements OnInit, OnChanges, OnDestroy {
+        return Math.round(this.scale * 100) + '%';
     }
 
-    private element: HTMLElement;
-
     constructor(
-        private sanitizer: DomSanitizer,
         private appConfigService: AppConfigService,
-        private contentService: ContentService,
-        private el: ElementRef) {
+        private contentService: ContentService) {
         this.initializeScaling();
     }
 
     initializeScaling() {
         const scaling = this.appConfigService.get<number>('adf-viewer.image-viewer-scaling', undefined) / 100;
         if (scaling) {
-            this.scaleX = scaling;
-            this.scaleY = scaling;
+            this.scale = scaling;
         }
     }
 
-    ngOnInit() {
-        this.element = <HTMLElement> this.el.nativeElement.querySelector('#viewer-image');
-
-        if (this.element) {
-            this.element.addEventListener('mousedown', this.onMouseDown.bind(this));
-            this.element.addEventListener('mouseup', this.onMouseUp.bind(this));
-            this.element.addEventListener('mouseleave', this.onMouseLeave.bind(this));
-            this.element.addEventListener('mouseout', this.onMouseOut.bind(this));
-            this.element.addEventListener('mousemove', this.onMouseMove.bind(this));
-        }
+    ngAfterViewInit() {
+        this.cropper = new Cropper(this.imageElement.nativeElement, {
+            autoCrop: false,
+            dragMode: 'move',
+            background: false,
+            scalable: true,
+            zoomOnWheel: false,
+            toggleDragModeOnDblclick: false,
+            viewMode: 1,
+            checkCrossOrigin: false,
+            ready: () => {
+                window.addEventListener('keydown', (this.onKeyDown.bind(this)));
+
+                if (this.imageElement.nativeElement.width < this.cropper.getContainerData().width) {
+                    const width = this.imageElement.nativeElement.width;
+                    const height = this.imageElement.nativeElement.height;
+                    const top = (this.cropper.getContainerData().height - this.imageElement.nativeElement.height) / 2;
+                    const left = (this.cropper.getContainerData().width - this.imageElement.nativeElement.width) / 2;
+
+                    this.cropper.setCanvasData({
+                        width,
+                        height,
+                        top,
+                        left
+                    });
+                }
+            }
+        });
     }
 
     ngOnDestroy() {
-        if (this.element) {
-            this.element.removeEventListener('mousedown', this.onMouseDown);
-            this.element.removeEventListener('mouseup', this.onMouseUp);
-            this.element.removeEventListener('mouseleave', this.onMouseLeave);
-            this.element.removeEventListener('mouseout', this.onMouseOut);
-            this.element.removeEventListener('mousemove', this.onMouseMove);
-        }
+        window.removeEventListener('keydown', this.onKeyDown);
```

Comment:
I'm not sure, it works. You are adding an event listener like with bind, like this:
```
window.addEventListener('keydown', (this.onKeyDown.bind(this)));
```

bind always return a new function.
And here, you try to remove not this already bound function, but simply the class method.

Reply:
One best practice, is to create a bound function in the constructor, and use this everywhere. This way, you are registering and unregistering the same function.
like:
```
 constructor(
        private appConfigService: AppConfigService,
        private contentService: ContentService) {
        this.initializeScaling();
        this.boundOnKeyDown = this.onKeyDown.bind(this);
    }
```

Reply:
updated with hostlistener, thanks for suggestion